### PR TITLE
Look for short hostname for foreman host

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: lib/foreman_rh_cloud.rb [hunk 1]
```diff
@@ -98,7 +98,18 @@ def self.transform_scheme(uri_string)
...
   # For testing purposes we can override the default hostname with an environment variable SATELLITE_RH_CLOUD_FOREMAN_HOST
   def self.foreman_host
-    @foreman_host ||= ::Host.unscoped.friendly.find(ENV['SATELLITE_RH_CLOUD_FOREMAN_HOST'] || ::SmartProxy.default_capsule.name)
+    @foreman_host ||= begin
+                        fullname = foreman_host_name
+                        ::Host.unscoped.friendly.find(fullname)
+                      rescue ActiveRecord::RecordNotFound
+                        # fullname didn't work. Let's try shortname
+                        shortname = /(?<shortname>[^\.]*)\.?.*/.match(fullname)[:shortname]
+                        ::Host.unscoped.friendly.find(shortname)
```
Comment (laviro) [inline lib/foreman_rh_cloud.rb:107]: in case the host's `shortname` wasn't found, should we log some error?

Comment (ShimShtein) [reply lib/foreman_rh_cloud.rb:107]: `find` will throw an exception in case it's not found.

Comment (laviro) [reply lib/foreman_rh_cloud.rb:107]: Great, thanks!

**Comment by laviro** [comment]
Thanks @ShimShtein !
do we need to CP it?

**Comment by ShimShtein** [comment]
CPs created

**Pull request merged**